pkgs/by-name/te/telegram-macos/package.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenvNoCC,
+  unzip,
+  fetchurl,
+}:
+let
+  version = "10.15.2-265540";
+  versionTokens = lib.split "-" version;
+  versionFirst = lib.head versionTokens;
+  versionSecond = lib.last versionTokens;
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit version;
+
+  pname = "telegram-macos";
+
+  src = fetchurl {
+    url = "https://osx.telegram.org/updates/Telegram-${versionFirst}.${versionSecond}.app.zip";
+    hash = "sha256-ne087xxFihsv+3UMFFk3zGzaweRG9o8mVwXkMIwH/9Y=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
```

Comment:
Is this necessary?  `fetchzip` seems like the more straightforward option and then we can disable the buildPhase altogether.